### PR TITLE
Add scalafmt to the repository

### PR DESCRIPTION
**Pull request by satorg**
This is an effort to introduce ScalaCheck and ScalaFmt to each other.
Recently I realized that those two great guys have never ever met.
So I decided to give it a try.

### CODE REVIEW

**Comment by satorg** [comment]
I've created as a Draft because before starting applying the formatting, I would like to agree on the scalafmt.conf file itself.
The initial scalafmt.conf is very minimalistic, but I am open to tune it as necessary to make everyone comfortable with it.

**Comment by vivekragunathan** [comment]
It would be nice if we agree to apply the formatting to `build.sbt` too. 

Also I think we should takes a few rounds of applying the formatting to reduce change history overload. Start with something small and iteratively apply other (bigger) things.

**Comment by satorg** [comment]
> It would be nice if we agree to apply the formatting to build.sbt too.

Yes sure – this PR implies formatting for build.sbt too.

**Comment by rossabaker** [comment]
Don't forget that format commits can be added to a .git-blame-ignore-revs file.  Most Typelevel repos use the same config, as a lot of the benefit is to avoid fights over inconsequential preferences.  I'm in favor of this, and I think one isolated commit to do the reformat, added to blame ignore, would be great.

**Comment by vivekragunathan** [comment]
> Don't forget that format commits can be added to a .git-blame-ignore-revs file. Most Typelevel repos use the same config, as a lot of the benefit is to avoid fights over inconsequential preferences. I'm in favor of this, and I think one isolated commit to do the reformat, added to blame ignore, would be great.

Oh wow, I did not that. That would work out great.

**Comment by rossabaker** [comment]
A really cool thing is that Scala Steward adds to that file whenever the formatting changes through plugin upgrades.

File: project/plugin.sbt [hunk 1]
```diff
@@ -3,6 +3,7 @@ addSbtPlugin("org.scala-native" % "sbt-scala-native" % "0.4.10")
 addSbtPlugin("org.portable-scala" % "sbt-scalajs-crossproject" % "1.2.0")
 addSbtPlugin("org.portable-scala" % "sbt-scala-native-crossproject" % "1.2.0")
 addSbtPlugin("pl.project13.scala" % "sbt-jmh" % "0.4.4")
+addSbtPlugin("org.scalameta" % "sbt-scalafmt" % "2.5.0")
 val sbtTypelevelVersion = "0.4.18"
 addSbtPlugin("org.typelevel" % "sbt-typelevel-ci-release" % sbtTypelevelVersion)
 addSbtPlugin("org.typelevel" % "sbt-typelevel-settings" % sbtTypelevelVersion)
```
Comment (armanbilge) [inline project/plugin.sbt:9]: Now that we have a scalafmt config, if we're willing to throw in copyright headers than we can upgrade to `sbt-typelevel` proper:
```suggestion
addSbtPlugin("org.typelevel" % "sbt-typelevel" % "0.4.18")
```

**Comment by rossabaker** [comment]
Also needs a `githubWorkflowGenerate`?

**Comment by satorg** [comment]
@rossabaker , I'm struggling with the `startYear` setting, which is required for sbt-header.
Currently in the headers it is set in this way:
```
/*-------------------------------------------------------------------------*\
**  ScalaCheck                                                             **
**  Copyright (c) 2007-2021 Rickard Nilsson. All rights reserved.          **
...
```
But if I set it to 2007 and keep using the "BSD-3-Clause" license, then the headers become
```
/*
 * Copyright (c) 2007, Typelevel
 * All rights reserved.
...
```
Would such a change be fine there?

**Comment by satorg** [comment]
I think this PR is ready for review.

Feel free to suggest changes to the `.scalafmt.conf` file please – I tried to be conservative in this initial work, but open to any other formatting rules.

Also, please pay attention to scaladoc comments – turned out they've changed the most after the re-formatting.

The last commit in the PR contains changes to `.git-blame-ignore-revs`.

Thank you.

**Review by rossabaker — APPROVED** [review]
This is good!  Sorry, it has been a chaotic week at $WORK and $HOME.

**Pull request merged**